analytics/analytics.go

```diff
@@ -28,15 +29,16 @@ import (
 // the name of the CI tool and true if we detect one.
 func DetectCI() (string, bool) {
 	for k, v := range map[string]string{
-		"GITHUB_WORKFLOW": "github-actions",
-		"CIRCLECI":        "circle-ci",
-		"JENKINS_HOME":    "jenkins",
-		"BUILDKITE":       "buildkite",
-		"DRONE_BRANCH":    "drone",
-		"TRAVIS":          "travis",
-		"GITLAB_CI":       "gitlab",
-		"EARTHLY_IMAGE":   "earthly-image",
-		"AGENT_WORKDIR":   "jenkins", // https://github.com/jenkinsci/docker-agent/blob/master/11/alpine/Dockerfile#L35
+		"GITHUB_WORKFLOW":        "github-actions",
+		"CIRCLECI":               "circle-ci",
+		"JENKINS_HOME":           "jenkins",
+		"BUILDKITE":              "buildkite",
+		"DRONE_BRANCH":           "drone",
+		"TRAVIS":                 "travis",
+		"GITLAB_CI":              "gitlab",
+		"EARTHLY_IMAGE":          "earthly-image",
+		"AGENT_WORKDIR":          "jenkins", // https://github.com/jenkinsci/docker-agent/blob/master/11/alpine/Dockerfile#L35
+		reserved.EarthlyCIRunner: "earthly-ci",
```

Comment:
once the `urfave/cli` flags is used, I would remove this, then just check higher up (wherever `app` is available), if `app.EarthlyCIRunner` is `true`.